libr/core/visual.c

```diff
@@ -4036,3 +4036,101 @@ R_API int r_core_visual(RCore *core, const char *input) {
 	r_cons_show_cursor (true);
 	return 0;
 }
+
+// TODO: move this to the table api
+R_API void r_core_visual_list(RCore *core, RList *list, ut64 seek, ut64 len, int width, int use_color) {
+	ut64 mul, min = -1, max = -1;
+	RListIter *iter;
+	ListInfo *info;
+	int j, i;
+	RIO *io = core->io;
+	width -= 80;
+	if (width < 1) {
+		width = 30;
+	}
+
+	r_list_foreach (list, iter, info) {
+		if (min == -1 || info->pitv.addr < min) {
+			min = info->pitv.addr;
+		}
+		if (max == -1 || info->pitv.addr + info->pitv.size > max) {
+			max = info->pitv.addr + info->pitv.size;
+		}
+	}
+	mul = (max - min) / width;
+	if (min != -1 && mul != 0) {
+		const char * color = "", *color_end = "";
+		i = 0;
+		r_list_foreach (list, iter, info) {
+			if (use_color && info->perm != -1) {
+				color_end = Color_RESET;
+				if ((info->perm & R_PERM_X) && (info->perm & R_PERM_W)) { // exec & write bits
+					color = r_cons_singleton()->context->pal.graph_trufae;
+				} else if ((info->perm & R_PERM_X)) { // exec bit
+					color = r_cons_singleton()->context->pal.graph_true;
+				} else if ((info->perm & R_PERM_W)) { // write bit
+					color = r_cons_singleton()->context->pal.graph_false;
+				} else {
+					color = "";
+					color_end = "";
+				}
+			} else {
+				color = "";
+				color_end = "";
+			}
+			if (io->va) {
+				io->cb_printf ("%03d%c %s0x%08"PFMT64x"%s |", i,
+						r_itv_contain (info->vitv, seek) ? '*' : ' ',
+						color, info->vitv.addr, color_end);
+			} else {
+				io->cb_printf ("%03d%c %s0x%08"PFMT64x"%s |", i,
+						r_itv_contain (info->pitv, seek) ? '*' : ' ',
+						color, info->pitv.addr, color_end);
+			}
+			for (j = 0; j < width; j++) {
+				ut64 pos = min + j * mul;
+				ut64 npos = min + (j + 1) * mul;
+				if (info->pitv.addr < npos && (info->pitv.addr + info->pitv.size) > pos)
+					io->cb_printf ("#");
+				else io->cb_printf ("-");
+			}
+			if (io->va) {
+				io->cb_printf ("| %s0x%08"PFMT64x"%s",
+					color, r_itv_end (info->vitv), color_end);	
+			} else {
+				io->cb_printf ("| %s0x%08"PFMT64x"%s %s",
+					color, r_itv_end (info->pitv), color_end);
+			}
+			if (info->perm != -1) {
+				io->cb_printf (" %s ", r_str_rwx_i (info->perm));
+			} else {
+				io->cb_printf ("     ", r_str_rwx_i (info->perm));
+			}
+			if (info->extra) {
+				io->cb_printf (" %4s ", info->extra);
+			} else {
+				io->cb_printf ("      ", info->extra);
+			}
+			if (info->name) {
```

Comment:
imho is better to just use one statemnt and use the ternary operator to change the arguments passed to a single format string that covers both cases. it will make the code shorter

Reply:
Okay will do this only